client/rest-high-level/src/main/java/org/elasticsearch/client/ml/filestructurefinder/FileStructure.java

```diff
@@ -0,0 +1,516 @@
+
+    static {
+        PARSER.declareInt(Builder::setNumLinesAnalyzed, NUM_LINES_ANALYZED);
+        PARSER.declareInt(Builder::setNumMessagesAnalyzed, NUM_MESSAGES_ANALYZED);
+        PARSER.declareString(Builder::setSampleStart, SAMPLE_START);
+        PARSER.declareString(Builder::setCharset, CHARSET);
+        PARSER.declareBoolean(Builder::setHasByteOrderMarker, HAS_BYTE_ORDER_MARKER);
+        PARSER.declareString((p, c) -> p.setFormat(Format.fromString(c)), FORMAT);
+        PARSER.declareString(Builder::setMultilineStartPattern, MULTILINE_START_PATTERN);
+        PARSER.declareString(Builder::setExcludeLinesPattern, EXCLUDE_LINES_PATTERN);
+        PARSER.declareStringArray(Builder::setColumnNames, COLUMN_NAMES);
+        PARSER.declareBoolean(Builder::setHasHeaderRow, HAS_HEADER_ROW);
+        PARSER.declareString((p, c) -> p.setDelimiter(c.charAt(0)), DELIMITER);
+        PARSER.declareString((p, c) -> p.setQuote(c.charAt(0)), QUOTE);
+        PARSER.declareBoolean(Builder::setShouldTrimFields, SHOULD_TRIM_FIELDS);
+        PARSER.declareString(Builder::setGrokPattern, GROK_PATTERN);
+        PARSER.declareString(Builder::setTimestampField, TIMESTAMP_FIELD);
+        PARSER.declareStringArray(Builder::setJodaTimestampFormats, JODA_TIMESTAMP_FORMATS);
+        PARSER.declareStringArray(Builder::setJavaTimestampFormats, JAVA_TIMESTAMP_FORMATS);
+        PARSER.declareBoolean(Builder::setNeedClientTimezone, NEED_CLIENT_TIMEZONE);
+        PARSER.declareObject(Builder::setMappings, (p, c) -> new TreeMap<>(p.map()), MAPPINGS);
+        PARSER.declareObject(Builder::setIngestPipeline, (p, c) -> p.mapOrdered(), INGEST_PIPELINE);
+        PARSER.declareObject(Builder::setFieldStats, (p, c) -> {
+            Map<String, FieldStats> fieldStats = new TreeMap<>();
+            while (p.nextToken() == XContentParser.Token.FIELD_NAME) {
+                fieldStats.put(p.currentName(), FieldStats.PARSER.apply(p, c));
+            }
+            return fieldStats;
+        }, FIELD_STATS);
+        PARSER.declareStringArray(Builder::setExplanation, EXPLANATION);
+    }
+
+    private final int numLinesAnalyzed;
+    private final int numMessagesAnalyzed;
+    private final String sampleStart;
+    private final String charset;
+    private final Boolean hasByteOrderMarker;
+    private final Format format;
+    private final String multilineStartPattern;
+    private final String excludeLinesPattern;
+    private final List<String> columnNames;
+    private final Boolean hasHeaderRow;
+    private final Character delimiter;
+    private final Character quote;
+    private final Boolean shouldTrimFields;
+    private final String grokPattern;
+    private final List<String> jodaTimestampFormats;
+    private final List<String> javaTimestampFormats;
+    private final String timestampField;
+    private final boolean needClientTimezone;
```

Comment:
`boolean` matches the [server side](https://github.com/elastic/elasticsearch/blob/6f46584380361d4fa52feebe404cf78a3074bf68/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/filestructurefinder/FileStructure.java#L161).  This one is `boolean` rather than `Boolean` because it's always present in the output.  The other flags are sometimes not set at all.